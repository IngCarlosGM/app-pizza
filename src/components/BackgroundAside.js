import React from 'react';

const BackgroundAside = () => {

    return (
        <div className="relative">
            <img 
                className="object-fit"
                src={`../assets/imagebkg.png`} 
                alt="Fondo negro de la ilustraciÃ³n" 
            />

            <img 
                className="center-vertical-horizontal-absolute size-image-pizza absolute"
                src={`../assets/Pizza.png`} 
                alt="Imagen de una Pizza"
            />

            <img 
                className="absolute logo-aside"
                src={`../assets/Logo.png`} 
                alt="Logo de Best pizza"
            />
        </div>
    )
}

export default BackgroundAside;
