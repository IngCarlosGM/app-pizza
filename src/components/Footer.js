import React from 'react';

export const Footer = () => {
    return (
        <footer className="space-between align-center">
            <div>
                <a 
                    href="https://facebook.com"
                    target="_blank"
                >
                    <img 
                        className="footer__icons img-responsive pointer"
                        src={`../assets/logo-facebook.png`} 
                        alt="Logo best pizza blanco" 
                    />
                </a>

                <a 
                    href="https://instagram.com"
                    target="_blank"
                >
                    <img 
                        className="footer__icons img-responsive pointer"
                        src={`../assets/logo-instagram.png`} 
                        alt="Logo best pizza blanco" 
                        />
                </a>
            </div>

            <a 
                href="https://mybestpizza.net/"
                target="_blank"
            >
                <img 
                    className="footer__icons img-responsive pointer"
                    src={`../assets/logo-best-pizza-white.png`} 
                    alt="Logo best pizza blanco" 
                />
            </a>
        </footer>
    )
}