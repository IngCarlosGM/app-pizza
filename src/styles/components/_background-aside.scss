.size-image-pizza {
    font-size: 50px;
    height: 4em;
    width: 4em;
}

.logo-aside {
    height: 4em;
    left: 5em;
    top: 30px;
    width: 5em;
}