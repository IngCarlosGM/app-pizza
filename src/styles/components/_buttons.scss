.btn-block {
    width: 100%;
}

.btn__primary {
    @include btn;
    background-color: $color-primary;
    border-radius: 2px;
    box-shadow: 2px 4px 4px 1px rgba($color: $color-third, $alpha: 0.3);
    
    &:disabled {
        background-color: lighten($color: $color-primary, $amount: 15);
    }

    &:hover { 
        background-color: darken($color: $color-primary, $amount: 15);
    }
}