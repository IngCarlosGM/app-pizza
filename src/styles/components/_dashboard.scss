.btn-responsive {
    width: 20%;
}

.container-dashboard {
    height: calc(100vh - 60px);
    padding-right: 20px;

    @media screen and (min-width: $breakpoint-large) {
        width: 80%;
    }

}

.container-item-store {
    height: 190px;
    width: 100%;

    &:hover { 
        background-color: darken($color: $color-white, $amount: 15);
    }

    @media (min-width: $breakpoint-tablet) and (orientation: portrait) {
        height: 150px;
    }

    @media (min-width: $breakpoint-landscape-large) and (orientation: landscape) {
        height: 150px;
    }
}

.container-item-store--without-hover {
    cursor: default;

    &:hover { 
        background-color: darken($color: transparent, $amount: 0);
    }
}

.grid-shop {
    display: grid;
    gap: 20px;
    grid-template-columns: 24% 24% 24% 24%;
    max-height: calc(100vh - 290px);

    @media (min-width: $breakpoint-landscape-tablet) and (orientation: landscape) { 
            grid-template-columns: 32% 32% 32%;
    }

    @media (min-width: $breakpoint-tablet) and (orientation: portrait) {
        grid-template-columns: 29.6% 29.6% 29.6%;
    }

    @media (min-width: $breakpoint-tablet) and (orientation: landscape) {
        grid-template-columns: 22% 22% 22% 22%;
    }

    @media (min-width: $breakpoint-desktop) and (orientation: portrait) {
        grid-template-columns: 32% 32% 32%;
    }
    
    @media (min-width: $breakpoint-large) and (orientation: portrait) {
        grid-template-columns: 30% 30% 30%;
    }
}

.img-item-store {
    max-height: 100%;
    max-width: 100%;
    object-fit: cover;
}

.item-store {
    border-radius: 8px;
    height: 140px;
    margin-bottom: 10px;

    @media (min-width: $breakpoint-tablet) and (orientation: portrait) {
        height: 100px;
    }

    @media (min-width: $breakpoint-landscape-large) and (orientation: landscape) {
        height: 100px;
    }
}

.not-store-founds {
    color: $color-third;
    font-size: 14px;
}

.title-item-store {
    @include truncate-one-line;
    font-size: 14px;
    color: $color-secondary;
}

.addres-item-store {
    @include truncate-one-line;
    font-size: 12px;
    color: $color-third;
}