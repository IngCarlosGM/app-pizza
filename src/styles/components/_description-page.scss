.description-page {
    color: $color-third;
    margin-top: 10px;
}