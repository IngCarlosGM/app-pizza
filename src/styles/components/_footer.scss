footer {
    background-color: $color-secondary;
    bottom: 0px;
    height: 60px;
    left: 30%;
    padding-right: 80px;
    padding-left: 50px;
    position: absolute;
    width: 70%;
}

.footer__icons {
    height: 24px;
    margin-right: 10px;
    width: 24px;
}