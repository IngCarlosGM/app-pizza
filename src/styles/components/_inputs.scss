.input {
    border: 1px solid $color-third;
    background-repeat: no-repeat;
    background-position: right 9px;
    background-position-x: calc(100% - 4px);
    background-size: 20px;
    border-radius: 4px;
    color: $color-third;
    height: 40px;
    padding-left: 15px;
    padding-right: 30px;

    &:focus {
        outline: $color-primary solid 2px;
        border: 0px transparent;
    }
}

.input--block {
    width: 100%;
}

.input--password {
    background-image: url('../../assets/icons/ic_contrasena.png');
}

.input--user {
    background-image: url('../../assets/icons/ic_usuario.png');
}