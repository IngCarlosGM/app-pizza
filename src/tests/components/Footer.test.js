import { shallow } from "enzyme";
import { Footer } from "../../components/Footer";

describe('Teting DescriptionPage component', () => {

    const wrapper = shallow(
        <Footer />
    )
   
    test('should render correctly', () => {
        expect( wrapper ).toMatchSnapshot();
    });

    test('should to have tow <a/>', () => {
        expect( wrapper.find('a').length ).toBe( 2 );
    });

    test('should to have three <img/>', () => {
        expect( wrapper.find('img').length ).toBe( 3 );
    });

    test('should to have a Link', () => {
        expect( wrapper.exists('Link') ).toBe( true );
    });
    
});
