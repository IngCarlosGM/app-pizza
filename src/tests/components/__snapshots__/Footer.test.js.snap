// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Teting DescriptionPage component should render correctly 1`] = `
<footer
  className="space-between align-center"
>
  <div>
    <a
      href="https://facebook.com"
      rel="noreferrer"
      target="_blank"
    >
      <img
        alt="Logo best pizza blanco"
        className="footer__icons img-responsive pointer"
        src="../assets/logo-facebook.png"
      />
    </a>
    <a
      href="https://instagram.com"
      rel="noreferrer"
      target="_blank"
    >
      <img
        alt="Logo best pizza blanco"
        className="footer__icons img-responsive pointer"
        src="../assets/logo-instagram.png"
      />
    </a>
  </div>
  <Link
    to="/"
  >
    <img
      alt="Logo best pizza blanco"
      className="footer__icons img-responsive pointer"
      src="../assets/logo-best-pizza-white.png"
    />
  </Link>
</footer>
`;
